### PR TITLE
Paths msmbuilder tpt

### PR DESCRIPTION
**Pull request by lgsmith**
I took the tpt.path code that enspara was calling from the msmbuilder codebase and added it into the path.py source file in the enspara code base. Although it appears to work, I've not tried to use this code in a project. It will remove msmbuilder as a dependency for setting up enspara, which is the primary reason I did this.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -100,3 +100,6 @@ ENV/
 
 # Mac file caches
 .DS_Store
+
+# PyCharm project settings
+.idea/
```
Comment (justinrporter) [inline .gitignore:105]: I think we probably don't need to commit this.

File: enspara/cluster/kcenters.py [hunk 2]
```diff
@@ -85,7 +85,7 @@ def fit(self, X, init_centers=None):
             Begin clustring with these centers as cluster centers.
         """
 
-        t0 = time.clock()
+        t0 = time.perf_counter()()
```
Comment (justinrporter) [inline enspara/cluster/kcenters.py:88]: Nice catch! Not related to TPT but a good change.

File: enspara/tpt/__init__.py [hunk 3]
```diff
@@ -3,3 +3,4 @@
 
 from .core import committors, mfpts
 from .tpt import reactive_fluxes, net_fluxes, reactive_populations
+from .path import paths, top_path
```
Comment (justinrporter) [inline enspara/tpt/__init__.py:6]: File should end in a newline.

**Review by justinrporter — CHANGES_REQUESTED** [review]
Thanks for taking the time to do this. Doing this seems eminently reasonable to me, as msmbuilder is now archived. 

Before we merge, could we also see if MSMBuilder has any tests for this code and transplant those as well (i.e. [here](https://github.com/msmbuilder/msmbuilder/blob/master/msmbuilder/tests/test_tpt.py))?

Thanks in advance for your contribution!

File: .gitignore (hunk 1 above)
Comment (lgsmith) [reply .gitignore:105]: I noticed that there were other varieties of editor project settings in the gitignore so I committed mine. Should the others go as well?

File: enspara/cluster/kcenters.py (hunk 2 above)
Comment (lgsmith) [reply enspara/cluster/kcenters.py:88]: I found a number of other instances of this substitution, and accidentally only included this one in this PR. I'm going to remove it here, and submit them all together as a separate PR, but I may have to include them as my python is to recent to use the time.clock function

File: enspara/tpt/__init__.py (hunk 3 above)
Comment (lgsmith) [reply enspara/tpt/__init__.py]: Yep fixed.

**Comment by lgsmith** [comment]
I've realized there is another reference to MSMBuilder I haven't expunged here yet, which is in _get_distance_method (it uses msmbuilder's libdistance to cover metrics that are not 'rmsd' or 'euclidean'. I'm afraid I'll have to try looking at that to see if it can be transplanted as well. Sorry to have sent this along prematurely.

**Comment by justinrporter** [comment]
> Sorry to have sent this along prematurely.

No trouble at all! Thanks for your time and contribution (in advance)!

> cover metrics that are not 'rmsd' or 'euclidean'.

Yes, I noticed that. I am not sure if anybody actually uses these. And, actually, I it's structured so you don't need the import unless you go down that branch. You could change the condition from something like:

```python
    if metric == 'rmsd':
        return md.rmsd
    if metric == 'euclidean':
        return euclidean
    elif isinstance(metric, str):
        try:
            import msmbuilder.libdistance as libdistance
        except ImportError:
            raise ImproperlyConfigured()
```

to something like

```python
    if metric == 'rmsd':
        return md.rmsd
    if metric == 'euclidean':
        return euclidean
    if metric in list_of_msmb_metrics:
        try:
            import msmbuilder.libdistance as libdistance
        except ImportError:
            raise ImproperlyConfigured("you need msmb, it's an optional dependency")
```

And then msmbuilder would truly be an _optional_ dependency.

Alternatively, you could implement/copy-paste the missing distances from msmbuilder.

Alternatively alternatively, I don't think anybody uses the more exotic distance metrics anyway, I think it would probably be fine if we deprecated them until/unless someone complains.

File: enspara/cluster/kcenters.py (hunk 2 above)
Comment (justinrporter) [reply enspara/cluster/kcenters.py:88]: > submit them all together as a separate PR

I love it.

**Comment by lgsmith** [comment]
These changes have been made. I wrote a more verbose help message in the libdistance error, but it is substantially similar to what you recommended. I'll try to submit the time pr next. It'll depend on this one...

**Comment by lgsmith** [comment]
What is missing at this point? I think I've addressed each change requested, though I may not have to @justinrporter 's satisfaction.

**Comment by justinrporter** [comment]
Have you run the tests? Unfortunately our test bot is broken at the moment; I was waiting to run the tests on my personal laptop but the new pip backtracking behavior is preventing me from installing your fork and running the tests :/

**Comment by justinrporter** [comment]
Got it working. LGTM

**Comment by lgsmith** [comment]
I actually found an issue with the test code where I hadn't correctly renamed the imports. Did it run OK for you?

**Pull request merged**